content/300-guides/300-upgrade-guides/200-upgrading-versions/600-upgrading-to-prisma-5.mdx

```diff
@@ -0,0 +1,677 @@
+
+You could also create a reusable error adapter which can be used in a `catch` in your code.
+
+```js
+const adaptError = (customThrowFn) => (error) => {
+  if (error.code === 'P2025') {
+    throw customThrowFn()
+  }
+  throw error
+}
+
+const user = await prisma.user.findFirstOrThrow({
+  where: { name: 'Alice' },
+}).catch(adaptError(() => new MyCustomError())
+```
+
+#### Replacing <inlinecode>rejectOnNotFound</inlinecode> enabled at the client level.
+
+For cases like these below:
+
+```js
+// Example 1
+const prisma = new PrismaClient({
+  rejectOnNotFound: true,
+})
+
+// Example 2
+const prisma = new PrismaClient({
+  rejectOnNotFound: {
+    findUnique: true,
+  },
+})
+```
+
+You will need to update your codebase to use `findUniqueOrThrow` and `findFirstOrThrow` instead of `findUnique` and `findFirst`, depending on which calls you would like to throw.
+
+If instead you use a custom error handler with the `rejectOnNotFound` property, like these examples:
+
+```js
+// Example 3
+const prisma = new PrismaClient({
+  rejectOnNotFound: (err) => new Error('something'),
+})
+
+// Example 4
+const prisma = new PrismaClient({
+  rejectOnNotFound: {
+    findUnique: (err) => new Error('something'),
+  },
+})
+
+// Example 5
+const prisma = new PrismaClient({
+  rejectOnNotFound: {
+    findFirst: {
+      User: (err) => new Error('User error'),
+      Post: (err) => new Error('Post error'),
+    },
+    findUnique: {
+      User: (err) => new Error('User error'),
+      Post: (err) => new Error('Post error'),
+    },
+  },
+})
+```
+
+You will need to update your method usage to `...OrThrow` and then use a [client extension](/concepts/components/prisma-client/client-extensions) in order to get the same behavior. As an example, the following extension would give the same behavior in Prisma 5 that `Example 5` gave in Prisma 4 and lower.
+
+```js
+import { PrismaClient } from '@prisma/client';
+
+const customErrorFunc = async (model, query, args) => {
+  try {
+    await query(args)
+  } catch (error: any) {
+    if (error.code === 'P2025') {
+      throw new Error(`${model} error`)
+    }
+    throw error;
+  }
+}
+
+const prisma = (new PrismaClient()).$extends({
+  query: {
+    user: {
+      async findFirstOrThrow({ model, query, args }) {
+        return await customErrorFunc(model, query, args)
+      },
+      async findUniqueOrThrow({ model, query, args }) {
+        return await customErrorFunc(model, query, args)
+      },
+    },
+    post: {
+      async findFirstOrThrow({ model, query, args }) {
+        return await customErrorFunc(model, query, args)
+      },
+      async findUniqueOrThrow({ model, query, args }) {
+        return await customErrorFunc(model, query, args)
+      },
+    },
+  },
+})
+```
+
+### <inlinecode>jsonProtocol</inlinecode> out of Preview!
+
+The `jsonProtocol` preview feature is now Generally Available. When upgrading to Prisma 5, make sure to remove `jsonProtocol` from your preview features, if added.
```

Comment:
Add a sentence to highlight the benefits and prepare to add a link to the blog post here.